nilearn/surface/surface.py

```diff
@@ -619,27 +619,56 @@ def _gifti_img_to_mesh(gifti_img):
             coords = gifti_img.get_arrays_from_intent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_POINTSET')
+            raise ValueError('The surf_mesh input is not recognized. '
```

Comment:
The error message you opted for has similarity with documentation. If user could not understand based on documentation then it is high likely chance that user may not understand from this error message.

Personally I would like something like this:
```python
The surf_mesh input is not recognized. Valid Freesurfer surface mesh inputs are ”pial” or “inflated” or “sphere” or “orig” or “white” which has intent code “NIFTI_INTENT_POINTSET”. You provided input which has no “NIFTI_INTENT_POINTSET” of value {0}.format(gifti_img.get_arrays_from_intent(nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET']))
```

Reply:
If you agree on this and need some help. We can go deep into this for a fix.

Reply:
I would like to have NIFTI_INTENT_POINTSET additionally in error message because it signifies that the vector value at each voxel is really a spatial coordinate (e.g., the vertices or nodes of a surface mesh)

Reply:
Yeah I will consider all those changes thanks for the help and I will work on it